src/module/producers.rs

```diff
@@ -113,3 +90,29 @@ impl ModuleProducers {
         })
     }
 }
+
+impl Module {
+    /// Parse a producers section from the custom section payload specified.
+    pub(crate) fn parse_producers_section(
+        &mut self,
+        mut data: wasmparser::BinaryReader,
```

Comment:
Seems like we should upstream a `ProducersSectionReader` or whatever to `wasmparser`